subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/AnnotationProcessorDetector.java

```diff
@@ -66,6 +66,9 @@ public FileCollection getEffectiveAnnotationProcessorClasspath(final CompileOpti
             if (pos == compileOptions.getCompilerArgs().size() - 1) {
                 throw new InvalidUserDataException("No path provided for compiler argument -processorpath in requested compiler args: " + Joiner.on(" ").join(compileOptions.getCompilerArgs()));
             }
+            DeprecationLogger.nagUserOfDeprecated(
```

Comment:
If we do this, we should un-incubate the `annotationProcessorPath` property (which is a good idea anyway, as the Android plugin already depends on it.

Reply:
It was suggested in https://github.com/gradle/gradle/pull/3058#discussion_r141895760 that maybe annotation processor-related compiler options could be refactored into a nested object (similar to what I did in `net.ltgt.apt` plugin, except at the task level, but I didn't have the choice has compile options has only become an extensible DSL in 4.3; with this specific PR linked above actually), so maybe it's a bit premature to un-incubate `annotationProcessorPath`? (your call)

Reply:
It's already used by one of the most important plugins out there, so we can't remove it without a deprecation period anyway. So it is not incubating in practice. We can deprecate it later if we move to a nested object.

Reply:
> It's already used by one of the most important plugins out there

I'm assuming you're talking about the Android plugin here (from earlier comment).
Fwiw, I've been told `net.ltgt.apt` is in the top 5 most downloaded plugins from the plugin portal; so that makes 2 “most important plugins” using `annotationProcessorPath` :wink:
(but `net.ltgt.apt` is versioned in a 0.x scheme, specifically because the goal was to have those things in Gradle proper eventually)

Reply:
Even more reason then :)